ai_diffusion/ui/autocomplete.py

```diff
@@ -0,0 +1,243 @@
+        tag_item = self._tag_item(index)
+
+        painter = ensure(painter)
+        painter.save()
+        if option.state & QStyle.StateFlag.State_Selected:
+            painter.fillRect(option.rect, option.palette.highlight())
+            text_color = option.palette.highlightedText().color()
+        else:
+            painter.fillRect(option.rect, self._background_color(tag_item.type_))
+            text_color = option.palette.text().color()
+
+        # Set up fonts
+        normal_font = QFont(option.font)
+        small_font = QFont(option.font)
+        small_font.setPointSize(normal_font.pointSize() - 2)
+        small_font.setItalic(True)
+
+        # Calculate rectangles
+        rect = option.rect
+        meta_width = QFontMetrics(normal_font).width(tag_item.meta) + 10
+        meta_rect = QRect(rect.right() - meta_width, rect.top(), meta_width, rect.height())
+
+        # Draw the tag
+        painter.setFont(normal_font)
+        painter.setPen(QPen(text_color))
+        painter.drawText(
+            rect, Qt.AlignmentFlag.AlignLeft | Qt.AlignmentFlag.AlignVCenter, tag_item.tag
+        )
+
+        # Draw the meta info
+        painter.setFont(small_font)
+        painter.setPen(QPen(text_color.lighter(150)))  # Slightly lighter color for meta info
+        painter.drawText(
+            meta_rect, Qt.AlignmentFlag.AlignRight | Qt.AlignmentFlag.AlignVCenter, tag_item.meta
+        )
+
+        painter.restore()
+
+    def sizeHint(self, option, index):
+        tag_item = self._tag_item(index)
+
+        normal_font = QFont(option.font)
+        small_font = QFont(option.font)
+        small_font.setPointSize(normal_font.pointSize() - 2)
+
+        tag_width = QFontMetrics(normal_font).width(tag_item.tag)
+        meta_width = QFontMetrics(small_font).width(tag_item.meta)
+
+        total_width = tag_width + meta_width + 10  # Add some padding
+        size = super().sizeHint(option, index)
+
+        return QSize(total_width, size.height())
+
+    def _tag_item(self, index) -> TagItem:
+        model = index.model()
+        while isinstance(model, QAbstractProxyModel):
+            model = model.sourceModel()
+        assert isinstance(model, TagListModel)
+        return model.tags[index.model().mapToSource(index).row()]
+
+    def _background_color(self, tag_type):
+        # danbooru tag color scheme
+        colors = {
+            TagType.General: QColor(230, 230, 255),  # Light blue
+            TagType.Artist: QColor(255, 230, 230),  # Light red
+            TagType.Copyright: QColor(255, 255, 230),  # Light pink
+            TagType.Character: QColor(230, 255, 230),  # Light green
+            TagType.Meta: QColor(230, 255, 255),  # Light yellow
+        }
+        return colors.get(tag_type, QColor(255, 255, 255))  # Default to white
+
+
+# Ensure there's only one of these globally. It gets pretty big if we have one per widget.
+TAG_MODEL = TagListModel([])
+TAG_FILES = None
```

Comment:
please rename to `_tag_model` and `_tag_files` (they're not constants and not public)

Reply:
done